### ISSUE TITLE
Seemingly bogus warning on bufferImageHeight w.r.t minImageTransferGranularity

### ISSUE THREAD

**Issue by haasn** [KhronosGroup/Vulkan-LoaderAndValidationLayers#2441]
https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers/blob/6203b40107017fb3f46aad3d136381810715e898/layers/buffer_validation.cpp#L1511

This check requires `VkBufferImageCopy.bufferImageHeight` to be a multiple of the queue family's `minImageTransferGranularity`. This behavior appears to be undocumented in the vulkan standard. The only mention of `minImageTransferGranularity` is in this section:
https://www.khronos.org/registry/vulkan/specs/1.0/man/html/VkQueueFamilyProperties.html

This section only requires the VkExtent3D / VkOffset3D to conform to the required rules, and makes no mention of bufferImageHeight. Also, the rule specifically allows setting it to be equal to the image height even if the image height is not a multiple. The way the validation check is written write now, something like this will fail, even though the spec should permit it:

```
VkImage example_img = /* create new 100x100 image */

VkBufferImageCopy foo = {
    .bufferOffset = 0,
    .bufferRowLength = 100,
    .bufferImageHeight = 100,
    .imageOffset = {0, 0, 0};
    .imageExtent = {100, 100, 1};
    .imageSubresource = {
        .aspectMask = VK_IMAGE_ASPECT_COLOR_BIT
        .layerCount = 1,
    },
};

vkCmdCopyBufferToImage(cmd, buffer, example_img, 1, &foo);
```

This results in a message not unlike this:
```
[vo/gpu/vulkan] vk [DS] 88: vkCmdCopyBufferToImage(): pRegion[0].bufferImageHeight (100) must be an even integer multiple of this command buffer's queue family image transfer granularity width (8). (obj 0x7fc708b3e540 (VkCommandBuffer), loc 0x52c)
```

This makes no sense, because the scenario of `bufferImageHeight` being equal to `imageOffset.y + imageExtent.height` is definitely expected to be supported. In fact, just commenting out the line that sets `bufferImageHeight` (so that it infers 100 from the image's height) makes the call succeed, even though the specification states that leaving `bufferImageHeight` as 0 is the same as setting it to the image's height. So naturally, they should at least both be treated the same way, even if you insist on including a check.

P.s. Looking at this same check I also noticed that it's comparing the height against `granularity.width` as well, which makes even less sense - if anything, it should be using `granularity.height`, no?

**Comment by mark-lunarg** [comment]
Duplicate of #2382.  @haasn, if you get a chance, see if pr #2439 takes care of these issues.

**Comment by mark-lunarg** [comment]
Closing as duplicate.  @haasn, please reopen if some of these old checks are still leaking through.
